main/models.py

```diff
@@ -53,6 +54,58 @@ class ServiceActivity(db.Model):
     specialist_rating = db.Column(db.Integer())
     customer_rating = db.Column(db.Integer())
 
+    confirmed = db.Column(db.Boolean(), default=False)
+
+    @classmethod
+    def get_or_create(cls,
+                      specialist,
+                      customer,
+                      service,
+                      start=None,
+                      end=None,
+                      description=None,
+                      specialist_rating=None,
+                      customer_rating=None):
+        """
+        Func for adding ServicesActivity
+
+        :param Specialist specialist: specialist
+        :param Customer customer: customer
+        :param Service service: service
+        :param int start: start
+        :param int end: end
+        :param str description: description
+        :param int specialist_rating: specialist_rating
+        :param int customer_rating: customer_rating
+        :return :
+        """
+        try:
+            activity = ServiceActivity.query.filter_by(
+                specialist=specialist, customer=customer, service=service).one()
+            created = False
+        except NoResultFound:
+            activity = ServiceActivity(
+                specialist=specialist, customer=customer, service=service)
+            session.add(activity)
+            session.commit()
+
+            if start:
+                activity.start = start
+            if end:
+                activity.end = end
+            if description:
+                activity.description = description
+            if specialist_rating and specialist_rating <= 5:
+                activity.specialist_rating = specialist_rating
+            if customer_rating and customer_rating <= 5:
+                activity.customer_rating = customer_rating
+
+            session.commit()
```

Comment:
Not a very good idea. What if you want to prevent adding ServiceActivity if exception thrown later ?